src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,104 @@
+package balancer
+
+import "math/rand"
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		q1 := dialers[i].ConsecSuccesses() - dialers[i].ConsecFailures()
+		q2 := dialers[j].ConsecSuccesses() - dialers[j].ConsecFailures()
+		return q1 > q2
+	}}
+}
+
+// Fastest strategy always pick the dialer with lowest recent average connect time
+func Fastest(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		return dialers[i].AvgDialTime() < dialers[j].AvgDialTime()
```

Comment:
One risk with this is that if we're not setting dial timeouts, one unlucky dial can really skew the statistics (for example if the network happens to be down when we try one dialer).  After the one really long dial, it will take a while for the average to recover.

Two potential solutions to this are:

1. Use a moving average over the last X period
2. Use the median instead of the average 

Median can actually be accomplished memory efficiently using a [sketching algorithm](http://research.neustar.biz/2013/09/16/sketch-of-the-day-frugal-streaming/).

This will not prevent me from approving this PR, it's just something I noticed that we could improve at some point if we want.